src/main/groovy/edu/oregonstate/mist/osuevents/db/EventsDAOWrapper.groovy

```diff
@@ -0,0 +1,103 @@
+package edu.oregonstate.mist.osuevents.db
+
+import edu.oregonstate.mist.osuevents.core.Event
+import groovy.transform.InheritConstructors
+
+import static java.util.UUID.randomUUID
+
+class EventsDAOWrapper {
+
+    private final EventsDAO eventsDAO
+
+    EventsDAOWrapper(EventsDAO eventsDAO) {
+        this.eventsDAO = eventsDAO
+    }
+
+    /**
+     * Get all events.
+     * @return
+     */
+    List<Event> getEvents() {
+        List<Event> events = eventsDAO.getEvents(null)
+
+        events.each {
+            it.instances = eventsDAO.getInstances(it.eventID)
+        }
+    }
+
+    /**
+     * Get single event by ID.
+     * @param eventID
+     * @return
+     */
+    Event getEventByID(String eventID) {
+        if (!eventID) {
+            throw new EventsDAOWrapperException("Cannot get an event with a null eventID.")
+        }
+
+        def event = eventsDAO.getEvents(eventID)
+
+        if (event) {
+            Event singleEvent = event?.get(0)
+            singleEvent.instances = eventsDAO.getInstances(eventID)
+
+            singleEvent
+        } else {
+            null
+        }
```

Comment:
You don't need an `if-else` condition block here since you have already used Groovy's null safe operator here. `singleEvent` will automatically be `null` if `event` is `null`: http://therealdanvega.com/blog/2013/08/20/groovys-null-safe-operator

Reply:
Yes, but I don't need to get the event instances unless an event was returned. `if (event)` will return true if `event` contains something. `event` will always be non-null though because the DAO will return an empty (but instantiated) list if no events were found.